finagle-core/src/main/scala/com/twitter/finagle/Dtab.scala

```diff
@@ -261,6 +262,7 @@ object Dentry {
       lazy val show: String = Path.showElem(buf)
     }
 
+    @tailrec
```

Comment:
Oh, exactly.
I've tried to fix it, and got compile error.

```
[error] /Users/zaneli/ws/finagle/finagle-core/src/main/scala/com/twitter/finagle/Dtab.scala:267: @tailrec annotated method contains no recursive calls
[error]       def apply(s: String): Label = Label(Buf.Utf8(s))
```

This addition seems to be a my mistake.
I've reverted it.